modules/runtime/nuxeo-launcher-commons/src/main/java/org/nuxeo/launcher/config/ConfigurationHolder.java

```diff
@@ -0,0 +1,246 @@
+/*
+ * (C) Copyright 2020 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Kevin Leturc <kleturc@nuxeo.com>
+ */
+
+package org.nuxeo.launcher.config;
+
+import static java.util.Objects.requireNonNull;
+
+import java.nio.file.InvalidPathException;
+import java.nio.file.Path;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.Set;
+import java.util.function.UnaryOperator;
+
+import org.nuxeo.common.Environment;
+import org.nuxeo.common.codec.CryptoProperties;
+
+/**
+ * Class used to hold the configuration for Nuxeo.
+ *
+ * @since 11.5
+ */
+public class ConfigurationHolder {
+
+    /** Path representing the Nuxeo Server home directory. Usually {@code /opt/nuxeo/server}. */
+    protected final Path home;
+
+    /** Path representing the Nuxeo Server configuration file. Usually {@code /opt/nuxeo/server/bin/nuxeo.conf}. */
+    protected final Path nuxeoConf;
+
+    /**
+     * Properties representing concatenation of {@code templates/nuxeo.defaults}, {@link System#getProperties()} and
+     * {@code nuxeo.defaults} from each templates.
+     */
+    protected final Properties defaultConfig;
+
+    /** Properties representing {@code nuxeo.conf} with a fallback on {@link #defaultConfig}. */
+    protected final CryptoProperties userConfig;
+
+    protected boolean configurable;
+
+    public ConfigurationHolder(Path home) {
+        this(home, home.resolve("bin").resolve("nuxeo.conf"));
+    }
+
+    public ConfigurationHolder(Path home, Path nuxeoConf) {
+        this.home = requireNonNull(home).toAbsolutePath();
+        this.nuxeoConf = requireNonNull(nuxeoConf).toAbsolutePath();
+        // initialize some defaults
+        var basicConfig = new Properties();
+        UnaryOperator<String> homeResolver = k -> getHomePath().resolve(k).toString();
+        UnaryOperator<String> runtimeResolver = k -> getRuntimeHomePath().resolve(k).toString();
```

Comment:
Why not regular member functions?

Reply:
Mainly because I can't make them static as I need the holder instance, thus making them as regular fields will cause IDE warns saying these can be a local variable.

Also, I didn't want to make them a class method, otherwise, they will be accessible to the whole package and I didn't want to expose a method returning a Path as a String.